### DISCUSSION TITLE
UNFCCC and ToC and Publication Manifest building

### DISCUSSION THREAD

**Discussion by mrchristian** [petermr/semanticClimate#42] [General]
### Table of contents naming conventions

Note I will take the discussion at some point and turn it into documentation.

This is my suggestion so far for the HTML files names:

[toc_toplevel_sum_ses.html](https://github.com/semanticClimate/unfccc/blob/main/toc_toplevel_sum_ses.html) - This ToC contains a summary of top level document sections, including UNFCCC Sessions

[toc_ses_dec_res.html](https://github.com/semanticClimate/unfccc/blob/main/toc_ses_dec_res.html) - This ToC contains UNFCCC - Sessions, Decisions, and Resolutions

[toc_dec_res_13_20_CMA_1.html](https://github.com/semanticClimate/unfccc/blob/main/toc_dec_res_13_20_CMA_1.html) - not needed at this stage

### Re: Question - M,ust we have a reading order in the manifest file publication.json

The answer so far appears to be - yes!

Reading the W3C standard for Publication Manifests it appears that it is needed and cannot use the ToC instead - https://www.w3.org/TR/pub-manifest/#default-reading-order

Also I tested it out by removing reading order and the publication would not render in Vivlio - reporting an error.

I will follow up on the issue with vivlio at some point - logged here https://github.com/semanticClimate/unfccc/issues/4

**Comment by mrchristian** [comment]
Here is the orginal Issue from the UNFCCC publication repo - https://github.com/semanticClimate/unfccc/issues/3

**Comment by mrchristian** [comment]
To confirm what was discussed today for a minimum intial manifest reading order and table of contents we will need the following:

1. A 'Default Reading Order' - https://github.com/semanticClimate/unfccc/issues/3 'Expressing the reading order as a simple list of URLs' see W3C example https://www.w3.org/TR/pub-manifest/#example-30-expressing-the-reading-order-as-a-simple-list-of-urls
2. A default 'Machine-Processable Table of Contents' https://www.w3.org/TR/pub-manifest/#app-toc-structure - the file we use for this is called [toc_toplevel_sum_ses.html](https://github.com/semanticClimate/unfccc/blob/main/toc_toplevel_sum_ses.html)
3. A ToC contains all UNFCCC - Sessions, Decisions, and Resolutions - the file we use for this is called [toc_ses_dec_res.html](https://github.com/semanticClimate/unfccc/blob/main/toc_ses_dec_res.html). **Note:** Not only will files for Decisions and Resolutions need generating, but also files for the Sessions will need generating using content from the documents supplied by UNFCCC.

**Comment by petermr** [comment]
I am breaking the problem down and will hopefully be able to write JSON and
HTML automatically

On Thu, Nov 30, 2023 at 12:46 PM Simon Worthington ***@***.***>
wrote:

> To confirm what was discussed today for a minimum intial manifest reading
> order and table of contents we will need the following:
>
>    1. A 'Default Reading Order' - semanticClimate/unfccc#3
>    <https://github.com/semanticClimate/unfccc/issues/3> 'Expressing the
>    reading order as a simple list of URLs' see W3C example
>    https://www.w3.org/TR/pub-manifest/#example-30-expressing-the-reading-order-as-a-simple-list-of-urls
>    2. A default 'Machine-Processable Table of Contents'
>    https://www.w3.org/TR/pub-manifest/#app-toc-structure - the file we
>    use for this is called toc_toplevel_sum_ses.html
>    <https://github.com/semanticClimate/unfccc/blob/main/toc_toplevel_sum_ses.html>
>    3. A ToC contains all UNFCCC - Sessions, Decisions, and Resolutions -
>    the file we use for this is called toc_ses_dec_res.html
>    <https://github.com/semanticClimate/unfccc/blob/main/toc_ses_dec_res.html>.
>    *Note:* Not only will files for Decisions and Resolutions need
>    generating, but also files for the Sessions will need generating using
>    content from the documents supplied by UNFCCC.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/semanticClimate/discussions/42#discussioncomment-7717435>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS5PKOYI2BM7AQXTWOLYHB533AVCNFSM6AAAAABABB5GHSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOMJXGQZTK>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by mrchristian** [comment]
I've made a write-up of the data needed to create a ToC in accornance with W3C Publication Manifest specicications. Please see the following issue #https://github.com/semanticClimate/cma3-test/issues/18 

> Data needed for ToC and Manifest creation

> What is being built? A [Table of Contents](https://w3c.github.io/wpub/#pub-table-of-contents) (ToC) to the specification of Publication Manifest (W3C Recommendation 10 November 2020) https://www.w3.org/TR/pub-manifest/

> In this example we are making a single ToC for a publication (Note: Publications can have multiple ToCs - for menus, short and long ToC at the start of a publication, and for specific chapters, etc.). The ToC will be used for a web and epub menu; for a Preamble ToC that will be output an multi-format: Responsive web, webpub (CSS Paged Media typesetting), PDF, PoD Print; and in the Publication Manifest for User Agent use.

**Comment by mrchristian** [comment]
FYI - we have the CSS and HTML changes worked out for the UN Climate (UN FCCC) ToC worked out - here is a WIP example, it will be placed back on the proper test document this afternoon, then tomorrow I will document. After this we will do two more things: 1. Wrap up the UN Climate docs as Books (covers, front matter, back matter, etc) 2. Make a different style so that the ToC function and overall presentation shows how you can switch layout with one click. - https://vivliostyle.vercel.app/#src=https://raw.githubusercontent.com/rqpe/css-typesetting-sandbox/main/publication.json
